### ISSUE TITLE
Should dropdims accept dims as a regular argument?

### ISSUE THREAD

**Issue by yurivish** [JuliaLang/julia#28906]
It feels a little redundant to say dims twice:

```
julia> dropdims([1 2 3], dims=(1,))
3-element Array{Int64,1}:
 1
 2
 3
```

Among the functions for reshaping arrays (`reshape`, `permutedims`, `dropdims`), `dropdims` is the only one that accepts `dims` as a keyword argument — the rest take dimensions as a positional argument.

```
julia> permutedims([1 2 3], (2, 1))
3×1 Array{Int64,2}:
 1
 2
 3
```

**Comment by jebej** [comment]
> Note: It's possible to make this change backwards-compatibly by accepting both the keyword and positional argument.

I agree, I think this should be the standard for a lot of these new keyword arguments.

Also see #28303 .

**Comment by mbauman** [comment]
Yeah, this was a choice that seemed much more obvious back when the function was still called `squeeze`. For many of the `dims=` keyword accepting functions, there was an ambiguity with other optional positional arguments. 

One thing specific to `dropdims` is its relation to reduction functions. We had wanted to allow `squeeze(sum, A, dims=1)` to perform the reduction and drop dimensions together in a slightly more terse API (#23500).  While it's not ambiguous, does `dropdims(f, A, 1)` still make sense? Would you still expect to see a `dims` keyword argument there given that you're effectively doing `dropdims(f(A, dims=1), 1)`?

**Comment by yurivish** [comment]
Maybe `dropdims(A, 1, reduce=f)`?

**Comment by JeffBezanson** [comment]
`dims` is used when you're specifying which dimensions to operate on. `reshape` is totally different; there you're passing sizes, not selecting dimensions. `permutedims` is closer, since the contents of the tuple have the same meaning, but is still a bit different since the order is so significant.

**Comment by aminya** [comment]
shouldn't `dropdims` be intelligent and drop the dimensions that are 1?

Say `A=[1 2 3 4]`, becomes a `1*4 Array`. `dropdims` should be able to drop 1st dimension without providing any keyword argument.
```julia
julia> A1
1×4 Array{Int64,2}:
 1  2  3  4

 dropdims(A1,dims=1)
4-element Array{Int64,1}:
 1
 2
 3
 4
```
proposed `dropdims`
```julia
 dropdims(A1)
4-element Array{Int64,1}:
 1
 2
 3
 4
```

**Comment by timholy** [comment]
Doing it automatically is a source of bugs (e.g., https://groups.google.com/d/msg/julia-users/wfEEvYIHVGs/caBXtuZulXgJ), and also not inferrable.

**Comment by aminya** [comment]
> 
> 
> Doing it automatically is a source of bugs (e.g., https://groups.google.com/d/msg/julia-users/wfEEvYIHVGs/caBXtuZulXgJ), and also not inferrable.

Although it is good to keep the code type stable, we shouldn't forget that **Julia is a dynamic language** and having such method will get the job done, especially if the array is constructed somewhere else and we need to squeeze the dimensions.
